htdocs/core/class/conf.class.php

```diff
@@ -394,6 +394,9 @@ public function setValues($db)
 								if ($modulename == 'propale') {
 									$modulename = 'propal';
 								}
+								if ($modulename == 'projet') {
```

Comment:
I suggest to move to the usage of https://github.com/Dolibarr/dolibarr/blob/55e201f63fb3bcaeced946bf8e4f3f59d004ce51/htdocs/core/lib/functions.lib.php#L209-L229 .

Also, this is another location (besides API) where I see supplierproposal so it may need to be added as a deprecated module name as well.

Reply:
I wam thinking of :

```php
if (isset(DEPRECATED_MODULE_MAPPING[$modulename]) {
   $modulename=DEPRECATED_MODULE_MAPPING[$modulename]);
}
```

isModEnabled already does that, but $modulename is also use to check the path for the modules.


https://github.com/Dolibarr/dolibarr/blob/55e201f63fb3bcaeced946bf8e4f3f59d004ce51/htdocs/core/lib/functions.lib.php#L237-L257

Reply:
checking for `project` must check `projet` ?

projects are back

Reply:
same pb with contract

Reply:
Apparently the mapping old/new is not fully completed.

```
grep -rP 'class .* extends DolibarrModules' htdocs 
```

I implemented a mapping table in the test, maybe it's best to make this table available to production code and use that to determine the class for a module name (if it's in the table), and the search method if it's not in the table.

https://github.com/Dolibarr/dolibarr/blob/55e201f63fb3bcaeced946bf8e4f3f59d004ce51/test/phpunit/CommonClassTest.class.php#L301-L304

Reply:
Using 'array_flip' the deprecated table can be converted to a table that can convert the 'new' names to the 'old' names that still seem to be used as class names.

Anyway, rather than copying the same conversions "all over the place", there should be (a) common method(s).